src/payment.js

```diff
@@ -0,0 +1,174 @@
+const cart = require('./cart');
+const cardApi = require('./card');
+const { isFunction, vaultRequest } = require('./utils');
+
+function methods(request) {
+  return {
+    async createElements(params = {}, checkoutId) {
+      const gateways = await vaultRequest('get', '/gateways');
+      if (gateways.error) {
+        throw new Error(gateways.error);
+      }
+      await render(request, params, checkoutId, gateways);
+    },
+
+    async create(orderId, amount, method, source) {
+      return request('post', `/payments`, {
+        order_id: orderId,
+        amount,
+        method,
+        source,
+      });
+    },
+  };
+}
+
+async function render(request, params, checkoutId, gateways = {}) {
+  if (gateways.braintree && gateways.paypal) {
+    const order = await cart.methods(request).getOrder(checkoutId);
+    if (!order) {
+      throw new Error('Order not found');
+    }
+    if (!(window.braintree && window.braintree.client)) {
+      await loadScript(
+        'braintree-web',
+        'https://js.braintreegateway.com/web/3.57.0/js/client.min.js',
+      );
+    }
+    if (!(window.braintree && window.braintree.paypalCheckout)) {
+      await loadScript(
+        'braintree-web-paypal-checkout',
+        'https://js.braintreegateway.com/web/3.57.0/js/paypal-checkout.min.js',
+      );
+    }
+    if (!window.paypal) {
+      await loadScript(
+        'paypal-sdk',
+        `https://www.paypal.com/sdk/js?client-id=${gateways.paypal.client_id}&merchant-id=${gateways.paypal.merchant_id}&vault=true`,
+      );
+    }
+    return await braintreePayPalButton(request, order, params, gateways);
+  } else if (gateways.stripe) {
+    if (!window.Stripe) {
+      await loadScript('stripe-js', 'https://js.stripe.com/v3/');
+    }
+    return await stripeElements(request, params, gateways);
+  }
+
+  throw new Error('Gateway elements are not implemented');
+}
+
+const loadScript = async (id, src) =>
+  new Promise((resolve) => {
+    const script = document.createElement('script');
+    script.id = id;
+    script.src = src;
+    script.async = true;
+    script.type = 'text/javascript';
+    script.addEventListener('load', () => resolve(), {
+      once: true,
+    });
+    document.head.appendChild(script);
+  });
+
+async function braintreePayPalButton(request, order, params, gateways) {
```

Comment:
@vstudenichnik-insoft since this is a client facing library we only have a `cart` here and not an `order` (which isn't created until a cart is submitted and then can't have payment info modified

Reply:
Done